### PR TITLE
Upgrade to Storybook 7 + Vite builder

### PR DESCRIPTION
**Pull request by daneah**
**This change:** (check at least one)

- [ ] Adds a new feature
- [ ] Fixes a bug
- [x] Improves maintainability
- [ ] Improves documentation
- [ ] Is a release activity

**Is this a breaking change?** (check one)

- [ ] Yes
- [x] No

**Is the:** (complete all)

- [x] Title of this pull request clear, concise, and indicative of the issue number it addresses, if any?
- [x] Test suite(s) passing?
- [x] Code coverage maximal?
- [ ] Changeset added?
- [ ] Component status page up to date?

**What does this change address?**

We want to continue moving along with Storybook, especially across major versions, to ensure we're able to take advantage of new features, faster builds, and more.

**How does this change work?**

We initially attempted to upgrade Storybook in place, but ran into difficult-to-debug issues after the initial codemod steps. Instead, I ultimately ripped Storybook out completely, installed it from scratch, then added needed pieces back in. In the process, I moved to the Vite builder instead of the Webpack builder for significantly improved local development speed.

**Additional context**

Of note:

* I had to add a React import to the couple of React-related files in `.storybook`, it seems like the JSX transform wasn't working for them. This might be failing once uploaded to Chromatic.

Things that aren't working:

* The theming is working for the Storybook UI, but isn't working within the docs content—maybe related to [this issue](https://github.com/storybookjs/storybook/issues/20704)?

### CODE REVIEW

File: packages/pharos/src/components/image-card/pharos-image-card.wc.stories.jsx [hunk 1]
```diff
@@ -47,12 +52,7 @@ export const Base = {
       ${items.map((item, index) => {
         return html` <li style="grid-column: span 2">
           <storybook-pharos-image-card id="card-${index}" title="Card Title" link="#">
-            <img
-              id="image-${index}"
-              src="./images/item-detail/${item.image}"
-              alt="Card Title ${index}"
-              slot="image"
-            />
+            <img id="image-${index}" src="${item.image}" alt="Card Title ${index}" slot="image" />
```
Comment (drewgingerich) [inline packages/pharos/src/components/image-card/pharos-image-card.wc.stories.jsx:55]: Tangential to this PR, is `index` being injected into these strings correctly? I thought that using back-ticks was necessary for the `${}` template syntax to work.

File: packages/pharos/src/pages/item-detail/wc/item-carousel/item-carousel-item.ts [hunk 2]
```diff
@@ -5,7 +5,7 @@ import type { Item } from '../../types';
 export const ItemCarouselItem = (item: Item): TemplateResult => html`
   <storybook-pharos-link href="#" subtle flex>
     <div class="item-detail-page__grid--carousel-item">
-      <img src="./images/item-detail/${item.image}" alt="${item.title}" />
+      <img src="${item.image}" alt="${item.title}" />
```
Comment (drewgingerich) [inline packages/pharos/src/pages/item-detail/wc/item-carousel/item-carousel-item.ts:8]: A question for many of these URL changes: what let's us get rid of prefixes such as `./images/item-detail/`?

**Comment by drewgingerich** [comment]
Overall lgtm, just have a few questions!

File: .storybook/react/main.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+const fullConfig = {
+  ...config,
+  framework: {
+    name: '@storybook/react-vite',
+    options: {},
+  },
+  stories: [
+    {
+      directory: '../../packages/pharos',
+      files: '**/*.@(react|docs|pages).stories.@(js|jsx|mdx|ts)',
```
[the remark this answers is not in the document]
Comment (daneah) [reply .storybook/react/main.ts:12]: Not that I know of for now; this didn't change from the previous.

File: packages/pharos/src/components/image-card/pharos-image-card.wc.stories.jsx (hunk 1 above)
Comment (daneah) [reply packages/pharos/src/components/image-card/pharos-image-card.wc.stories.jsx:55]: Note the backticks several lines up where the `html` tagged literal opens!

File: packages/pharos/src/pages/item-detail/wc/item-carousel/item-carousel-item.ts (hunk 2 above)
Comment (daneah) [reply packages/pharos/src/pages/item-detail/wc/item-carousel/item-carousel-item.ts:8]: This and I think one or two other files have a layer of indirection making it not obvious from the changes here; `item` comes from an object in another module, and each of those items' `image` was changed from a string to an imported image file.

**Review by drewgingerich — APPROVED** [review]

File: .storybook/react/main.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (daneah) [reply .storybook/react/main.ts:12]: Oof, you were right about this! `tsx` got mucked up to `ts` in this changeset. Thanks for looking out.

**Pull request merged**